### DISCUSSION TITLE
Some regex query returned invalid even if it's not

### DISCUSSION THREAD

**Discussion by r0ny123** [community/community#9185] [Other Feature Feedback, Questions, & Ideas] [answered]
I tried following regex `/uint(8|16|32)\(@[^]*\[/` in code search but it returned saying invalid one though it's valid (https://regexr.com/6c0s7). Any explanation?

![image](https://user-images.githubusercontent.com/49360849/146979216-4e37ac87-542a-4613-be44-d245aecc5140.png)

**Comment by svick** [comment, marked as answer]
It's a valid regex in JS, but each regex engine is a bit different. Even if you switch the engine on RegExr, it will give you an error (it seems you can't have an empty negated character set).

Though it would be nice if GitHub would produce a better error.

**Comment by r0ny123** [comment]
Hello, thanks or pointing out! Maybe someone from @github pick this up.
